daemon/container_operations_unix.go

```diff
@@ -760,10 +764,6 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 	controller := daemon.netController
 
 	if endpointConfig != nil {
```

Comment:
If I read the code correctly, correct me if I am wrong, the check on alias can stay in this function, because in `runconfig.Parse()` you are discarding the options in case it is not a user-defined network.
In fact even today the code does not throw an error if alias is specified and container is run on the default network.